### PR TITLE
Downgrade android support libs to v23

### PR DESCRIPTION
**Pull request by LornTao**
@zugaldia @kkaefer 
Build fail, and got an error on NDK configuring:
```
"Install Android SDK Build-Tools 19.1" complete.
Incremental java compilation is an incubating feature.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':MapboxGLAndroidSDK'.
> NDK not configured. 
  Download it with SDK manager.)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```
How to modify the mason version , and fix this issue?

### CODE REVIEW

**Comment by zugaldia** [comment]
>  Submodule .mason added at e697e6

@LornTao You shouldn't need to change the version of Mason needed, or was this necessary?

File: platform/android/MapboxGLAndroidSDK/build.gradle [hunk 1]
```diff
@@ -16,9 +16,10 @@ dependencies {
     }
 
     // Mapbox Android Services (Telemetry support)
-    compile(rootProject.ext.dep.mapboxAndroidTelemetry) {
-        transitive = true
-    }
+    compile(name: 'libandroid-telemetry', ext: 'aar')
```
Comment (zugaldia) [inline platform/android/MapboxGLAndroidSDK/build.gradle]: If you're changing the telemetry library too, we'll need the equivalent PR on `mapbox-java`.

File: platform/android/MapboxGLAndroidSDK/gradle.properties [hunk 2]
```diff
@@ -18,3 +18,4 @@ POM_PACKAGING=aar
 # Only build native dependencies for the current ABI
 # See https://code.google.com/p/android/issues/detail?id=221098#c20
 android.buildOnlyTargetAbi=true
+org.gradle.jvmargs=-Xmx1536M
```
Comment (zugaldia) [inline platform/android/MapboxGLAndroidSDK/gradle.properties]: Why is this necessary?

File: .idea/encodings.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (zugaldia) [inline .idea/encodings.xml]: Let's remove any `.idea` files from this PR.

File: platform/android/MapboxGLAndroidSDK/build.gradle [hunk 4]
```diff
@@ -106,6 +107,9 @@ android {
     sourceSets {
         // limit amount of exposed library resources
         main.res.srcDirs += 'src/main/res-public'
+        main {
+            jni.srcDirs = []
+        }
```
Comment (kkaefer) [inline platform/android/MapboxGLAndroidSDK/build.gradle]: Does this mean that Android Studio won't compile the C++ code anymore?

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -11,8 +11,6 @@
         android:fullBackupContent="true"
         android:icon="@drawable/icon"
         android:label="@string/app_name"
-        android:roundIcon="@drawable/ic_launcher_round"
-        android:supportsRtl="true"
```
Comment (kkaefer) [inline platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml]: Why does this remove RTL support?

File: platform/android/build.gradle [hunk 6]
```diff
@@ -13,11 +13,14 @@ allprojects {
...
         maven { url "http://oss.sonatype.org/content/repositories/snapshots/" }
+        flatDir {
+            dirs '../MapboxGLAndroidSDK/libs'
+        }
     }
 }
 
 task wrapper(type: Wrapper) {
-    gradleVersion = '3.2.1'
+    gradleVersion = '2.14.1'
```
Comment (kkaefer) [inline platform/android/build.gradle]: Gradle 3 contains significant bug fixes that we require to build Mapbox GL as part of the Gradle build process. We can't downgrade to Gradle 2.

**Comment by zugaldia** [comment]
Continues on https://github.com/mapbox/mapbox-gl-native/pull/9742.
